ts/components/SelectCalendarModal.tsx

```diff
@@ -0,0 +1,108 @@
+import * as pot from "italia-ts-commons/lib/pot";
+import { Button, Container, Content, Text } from "native-base";
+import React from "react";
+import { FlatList, ListRenderItem, Modal } from "react-native";
+import RNCalendarEvents, { Calendar } from "react-native-calendar-events";
+import I18n from "../i18n";
+
+type CalendarItemProps = {
+  calendar: Calendar;
+  onPress: (calendar: Calendar) => void;
+};
+
+/**
+ * A component to render a Calendar as FlatList item
+ */
+class CalendarItem extends React.PureComponent<CalendarItemProps> {
+  public render() {
+    return <Text onPress={this.onPress}>{this.props.calendar.title}</Text>;
+  }
+
+  private onPress = () => {
+    const { calendar, onPress } = this.props;
+
+    onPress(calendar);
+  };
+}
+
+type Props = {
+  onCancel: () => void;
+  onCalendarSelected: (calendar: Calendar) => void;
+  header?: React.ReactNode;
+};
+
+type State = {
+  calendars: pot.Pot<ReadonlyArray<Calendar>, Error>;
+};
+
+const calendarKeyExtractor = (item: Calendar, _: number) => item.id;
+
+/**
+ * A modal that allow the user to select one of the device available Calendars
+ */
+// tslint:disable-next-line:max-classes-per-file
+class SelectCalendarModal extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      calendars: pot.none
+    };
+  }
+
+  public render() {
+    const { calendars } = this.state;
+
+    return (
+      <Modal visible={true} onRequestClose={this.props.onCancel}>
```

Comment:
do not use react native modals, they interfere with the identification screen

Reply:
I also don't like Modal, expecially in this case, but it is what the story is asking for:
```
Nell'aggiunta di un promemoria è sufficiente mostrare all'utente, in **modale**, la scelta di quale calendario deve ricevere i promemoria, fra quelli disponibili sul dispositivo.
```

Before changing the implementation i think that @matteodesanti need to decide a proper UX to avoid the need to rechange it.

What you think?

Reply:
Don't take what @matteodesanti said literally: @matteodesanti wants a "modal" like UI component (i.e. a temp content window that covers the screen), @matteodesanti doesn't refer to React Native's `Modal` component. 

We have an [Overlay](https://github.com/teamdigitale/italia-app/blob/master/ts/components/ui/Overlay.tsx) component that does pretty much what you need. See an example usage in [withErrorModal HOC](https://github.com/teamdigitale/italia-app/blob/master/ts/components/helpers/withErrorModal.tsx#L72). I guess the only thing you may want to change is that `Overlay` centers the content in the "modal", feel free to make that behavior configurable.

Reply:
The Overlay component is not a proper "modal" and seem not to work in my case. It works if you are wrapping the entire screen with the Overlay component, but i am in a deeper component so when you use the Overlay component it takes only the space of the deeper component.

![overlay](https://user-images.githubusercontent.com/30595520/51893088-2f7f6e80-23a4-11e9-9114-214762d91642.png)

Reply:
Going to try another way.

Reply:
Would you mind sharing it before implementing it? :)

Reply:
What i want to try is to create a ModalContext, ModalProvider and ModalConsumer than wrap the whole App with the ModalProvider (like we do here for the PopupMenu https://github.com/teamdigitale/italia-app/blob/master/ts/App.tsx#L29). In this way we can create a ModalRoot component at the RootContainer level and than with the ModalConsumer use the ModalProvider methods (show and hide that i'll create) to show something in the ModalRoot.

Reply:
With Context, Provider, Consumer i mean the new React 16.3 ContextApi.

Reply:
ok give it a try

Reply:
Done. Now we are using the LightModal instead of the react-native Modal.